File: src/main/java/com/kishlaly/ta/utils/Numbers.java
package com.kishlaly.ta.utils;

import java.math.BigDecimal;

public class Numbers {

    public static double round(double value) {
        if (Double.isNaN(value) || value == 0) {
            return 0d;
        }
        return BigDecimal.valueOf(value).setScale(2, BigDecimal.ROUND_UP).doubleValue();
    }

    public static int roundDown(double value) {
        if (Double.isNaN(value) || value == 0) {
            return 0;
        }
        return BigDecimal.valueOf(value).setScale(0, BigDecimal.ROUND_DOWN).intValue();
    }

    public static double roi(double initialCost, double currentCost) {
        double result = (currentCost - initialCost) / initialCost * 100;
        return round(result);
    }

    public static double percent(double n, double N) {
        return round(n * 100 / N);
    }

}

File: src/main/kotlin/com/kishlaly/ta/utils/Numbers.kt
package com.kishlaly.ta.utils

class Numbers {

    companion object {
        fun roi(initialCost: Double, currentCost: Double): Double {
            val result = (currentCost - initialCost) / initialCost * 100
            return result.round()
        }

        fun percent(n: Double, N: Double): Double {
            return (n * 100 / N).round()
        }

    }

}